AwsEncryptionSDK/runtimes/net/Examples/Keyring/AwsKmsHierarchical/AwsKmsHierarchicalKeyring.cs

```diff
@@ -51,158 +58,121 @@
             KmsClient = new AmazonKeyManagementServiceClient(),
-            KmsConfiguration = kmsConfig,
+            KmsConfiguration = new KMSConfiguration{ KmsKeyArn = ExampleUtils.ExampleUtils.GetBranchKeyArn() },
             DdbTableName = ExampleUtils.ExampleUtils.GetKeyStoreName(),
             DdbClient = new AmazonDynamoDBClient(),
             LogicalKeyStoreName = ExampleUtils.ExampleUtils.GetLogicalKeyStoreName() 
         };
-        var materialProviders = new MaterialProviders(new MaterialProvidersConfig());
-        var keystore = new KeyStore(keystoreConfig);
+        var keyStore = new KeyStore(keyStoreConfig);
+        
+        // 2. Create the DynamoDb table that will store the branch keys and beacon keys.
+        //    This checks if the correct table already exists at `ddbTableName`
+        //    by using the DescribeTable API. If no table exists,
+        //    it will create one. If a table exists, it will verify
+        //    the table's configuration and will error if the configuration is incorrect.
+        //    It may take a couple minutes for the table to become ACTIVE,
+        //    at which point it is ready to store branch and beacon keys.
+        
+        // keyStore.CreateKeyStore(new CreateKeyStoreInput());
+        
+        // We have already created a Table with the specified configuration.
+        // For testing purposes we will not create a table when we run this example.
         
-        // Create a branch key supplier that maps the branch key id to a more readable format
-        var branchKeySupplier = new ExampleBranchKeySupplier(branchKeyIdA, branchKeyIdB);
+        // 3. Create a new branch key and beacon key in our KeyStore.
+        //    Both the branch key and the beacon key will share an Id.
+        //    This creation is eventually consistent. See the CreateBranchKey
+        //    Example for how to populate this table.
         
-        // Create an AWS Hierarchical Keyring with the branch key id supplier
+        // var branchKeyId = CreateBranchKey.createBranchKey();
+        
+        // For testing purposes we will not create a table when we run this example.
+        // We will use an existing branch key created using the above code to run this
+        // example.
+        
+        // THIS IS A PUBLIC RESOURCE DO NOT USE IN A PRODUCTION ENVIRONMENT
+        var branchKeyId = "43574aa0-de30-424e-bad4-0b06f6e89478";
+        
+        // 4. Create the Hierarchical Keyring, using the Branch Key ID Supplier above.
+        //    With this configuration, the AWS SDK Client ultimately configured will be capable
+        //    of encrypting or decrypting items for either tenant (assuming correct KMS access).
+        //    If you want to restrict the client to only encrypt or decrypt for a single tenant,
+        //    configure this Hierarchical Keyring using `BranchKeyId = tenant1BranchKeyId` instead
+        //    of `BranchKeyIdSupplier = branchKeySupplier`.
```

Comment:
Question: What Branch Key ID Supplier?
I do not see one above, nor do I see one below.

Reply:
Copy paste fail